server.js

```diff
@@ -8,12 +9,63 @@ app.use(cors());
 
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
+const { request, response } = require("express");
 
 app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
+app.get("/bookings", (request, response) => {
+  response.json({ bookings });
+});
+
+app.get("/bookings/:id", (request, response) => {
+  const id = Number(request.params.id);
+  const findId = bookings.find((bookings) => bookings.id === id);
+  if (!findId) {
+    return response
+      .status(404)
+      .json({ message: `booking with the ${id} not found` });
+  }
+  response.status(200).json({ findId })
+});
+
+app.post("/bookings", (request, response) => {
+  if (
+    request.body.firstName === "" ||
```

Comment:
you could always do some destructuring here to avoid having to type request.body, for example:
`const {body} = request`